src/ccs/multifolding.rs

```diff
@@ -0,0 +1,390 @@
+use super::cccs::{CCCSInstance, CCCSShape};
```

Comment:
Should we just call this file nifs.rs to make it consistent with the existing code?